### PR TITLE
Add usage of benchmark runner

### PR DESCRIPTION
**Pull request by rashidnhm**
**Context:** The benchmarks workflow now use the self-hosted benchmark runner

**Description of the Change:**
The GitHub hosted runner is not very consistent with collecting accurate times. A dedicated benchmarking runner has been deployed in AWS and will now be used to run benchmarks.

**Benefits:**
Consistent times.

**Possible Drawbacks:**
Slower overall runtime as it can only process 1 job at a time.

**Related GitHub Issues:**

### CODE REVIEW

**Comment by rashidnhm** [comment]
> Hey @rashidnhm, thank you for that. This will be very important to keep track of PennyLane performance. I can see that we are running the benchmarks suite in a dedicated runner in AWS, but is this always the same machine, or are they equivalent machines? I'm asking that, because analyzing the last three runs, I can see that results are less erratic but still not very consistent. You can see that by comparing the images inside of artifacts with same name in different runs. Also, I can see that we are getting several warnings regarding Node.js 16 actions being deprecated. Can we have actions/checkout@v3, actions/cache@v3, actions/setup-python@v4, actions/upload-artifact@v3, upgraded to actions/checkout@v4, actions/cache@v4, actions/setup-python@v5, actions/upload-artifact@v4?

Hey Amintor, they are running on the exact same instance each time. It is a long running EC2 instance.

As for the deprecation warnings, that is a GitHub raised warning as the actions we use depend on a version of node that's need deprecated. Look at [this run](https://github.com/PennyLaneAI/pennylane/actions/runs/7656472654) of our test suite run on the master branch on github hosted runners. The same warnings are there. These warnings should be addressed but are out of scope for this work.

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -134,6 +134,22 @@ jobs:
...
+          rm -rf venv_*
+          python -m venv "$VENV_NAME"
+          echo "${{ github.workspace }}/$VENV_NAME/bin" >> $GITHUB_PATH
+
+      - name: Validate Python
+        if: steps.continue.outputs.confirm == 'true'
+        run: |
+          which python
+          which pip
+          python --version
```
[the remark this answers is not in the document]
Comment (AmintorDusko) [reply .github/workflows/unit-test.yml:151]: I agree. Thanks for clarifying.

**Review by AmintorDusko — APPROVED** [review]
Thank you @rashidnhm, I consider your work here nicely done. Upgrading actions can be done in a follow-up work. Regarding the results inconsistencies, I think this is work for another moment. Thank you very much.

**Comment by AmintorDusko** [comment]
Hey @Rashid, just flying by, I didn't check if you did it already, but it would be great if you could clear the reference-benchmarks cache and re-run the CIs to compare results from the same runner. Thanks.

**Comment by rashidnhm** [comment]
Sure

**Pull request closed without merging**